### ISSUE TITLE
Installation instructions issue?

### ISSUE THREAD

**Issue by tylergannon** [tramhao/termusic#90]
Hi, wondering if it's possible that there are some requirements missing from the README, or something else.  Pretty new to Rust development so I don't know how to approach the problem when the error message doesn't turn up any google results.

I've got a fresh installation of the 1.64.0 toolchain, running on up to date MacBook.  Compilation of all the deps succeeds.  Then during the final step (compiling termusic v0.7.3), I get the following.

I'm excited to try this tool, and would appreciate any pointers you might have.

```
error[E0432]: unresolved import `crate::DefaultStreamConfigError`
 --> /Users/tyler/.asdf/installs/rust/1.64.0/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/mod.rs:8:5
  |
8 | use crate::DefaultStreamConfigError;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `DefaultStreamConfigError` in the root

error[E0432]: unresolved imports `crate::BuildStreamError`, `crate::SupportedStreamConfigsError`
 --> /Users/tyler/.asdf/installs/rust/1.64.0/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/mod.rs:9:13
  |
9 | use crate::{BuildStreamError, SupportedStreamConfigsError};
  |             ^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `SupportedStreamConfigsError` in the root
  |             |
  |             no `BuildStreamError` in the root

error[E0432]: unresolved imports `crate::BackendSpecificError`, `crate::SampleFormat`, `crate::StreamConfig`
  --> /Users/tyler/.asdf/installs/rust/1.64.0/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/mod.rs:11:13
   |
11 | use crate::{BackendSpecificError, SampleFormat, StreamConfig};
   |             ^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^  ^^^^^^^^^^^^ no `StreamConfig` in the root
   |             |                     |
   |             |                     no `SampleFormat` in the root
   |             no `BackendSpecificError` in the root

error[E0432]: unresolved import `crate::traits`
  --> /Users/tyler/.asdf/installs/rust/1.64.0/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/macos/mod.rs:23:12
   |
23 | use crate::traits::{DeviceTrait, HostTrait, StreamTrait};
   |            ^^^^^^
   |            |
   |            unresolved import
   |            help: a similar path exists: `crate::player::rusty_backend::cpal::traits`

error[E0432]: unresolved imports `crate::BackendSpecificError`, `crate::BufferSize`, `crate::BuildStreamError`, `crate::ChannelCount`, `crate::Data`, `crate::DefaultStreamConfigError`, `crate::DeviceNameError`, `crate::DevicesError`, `crate::InputCallbackInfo`, `crate::OutputCallbackInfo`, `crate::PauseStreamError`, `crate::PlayStreamError`, `crate::SampleFormat`, `crate::SampleRate`, `crate::StreamConfig`, `crate::StreamError`, `crate::SupportedBufferSize`, `crate::SupportedStreamConfig`, `crate::SupportedStreamConfigRange`, `crate::SupportedStreamConfigsError`
  --> /Users/tyler/.asdf/installs/rust/1.64.0/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/macos/mod.rs:25:5
   |
25 |     BackendSpecificError, BufferSize, BuildStreamError, ChannelCount, Data,
   |     ^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^  ^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^  ^^^^ no `Data` in the root
   |     |                     |           |                 |
   |     |                     |           |                 no `ChannelCount` in the root
   |     |                     |           no `BuildStreamError` in the root
   |     |                     no `BufferSize` in the root
   |     no `BackendSpecificError` in the root
26 |     DefaultStreamConfigError, DeviceNameError, DevicesError, InputCallbackInfo, OutputCallbackInfo,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^  ^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^ no `OutputCallbackInfo` in the root
   |     |                         |                |             |
   |     |                         |                |             no `InputCallbackInfo` in the root
   |     |                         |                no `DevicesError` in the root
   |     |                         no `DeviceNameError` in the root
   |     no `DefaultStreamConfigError` in the root
27 |     PauseStreamError, PlayStreamError, SampleFormat, SampleRate, StreamConfig, StreamError,
   |     ^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^  ^^^^^^^^^^^^  ^^^^^^^^^^  ^^^^^^^^^^^^  ^^^^^^^^^^^
28 |     SupportedBufferSize, SupportedStreamConfig, SupportedStreamConfigRange,
   |     ^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^^^
29 |     SupportedStreamConfigsError,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0432]: unresolved imports `crate::BackendSpecificError`, `crate::DevicesError`, `crate::SupportedStreamConfigRange`
  --> /Users/tyler/.asdf/installs/rust/1.64.0/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/macos/enumerate.rs:11:13
   |
11 | use crate::{BackendSpecificError, DevicesError, SupportedStreamConfigRange};
   |             ^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^^^ no `SupportedStreamConfigRange` in the root
   |             |                     |
   |             |                     no `DevicesError` in the root
   |             no `BackendSpecificError` in the root

error[E0412]: cannot find type `HostUnavailable` in the crate root
  --> /Users/tyler/.asdf/installs/rust/1.64.0/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/macos/mod.rs:53:41
   |
53 |     pub fn new() -> Result<Self, crate::HostUnavailable> {
   |                                         ^^^^^^^^^^^^^^^ not found in the crate root
   |
help: consider importing this struct
   |
4  | use crate::player::rusty_backend::cpal::HostUnavailable;
   |
help: if you import `HostUnavailable`, refer to it directly
   |
53 -     pub fn new() -> Result<Self, crate::HostUnavailable> {
53 +     pub fn new() -> Result<Self, HostUnavailable> {
   |

error[E0422]: cannot find struct, variant or union type `InputStreamTimestamp` in the crate root
   --> /Users/tyler/.asdf/installs/rust/1.64.0/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/macos/mod.rs:683:36
    |
683 |             let timestamp = crate::InputStreamTimestamp { callback, capture };
    |                                    ^^^^^^^^^^^^^^^^^^^^ not found in the crate root
    |
help: consider importing this struct
    |
4   | use crate::player::rusty_backend::cpal::InputStreamTimestamp;
    |
help: if you import `InputStreamTimestamp`, refer to it directly
    |
683 -             let timestamp = crate::InputStreamTimestamp { callback, capture };
683 +             let timestamp = InputStreamTimestamp { callback, capture };
    |

error[E0422]: cannot find struct, variant or union type `OutputStreamTimestamp` in the crate root
   --> /Users/tyler/.asdf/installs/rust/1.64.0/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/macos/mod.rs:775:36
    |
775 |             let timestamp = crate::OutputStreamTimestamp { callback, playback };
    |                                    ^^^^^^^^^^^^^^^^^^^^^ not found in the crate root
    |
help: consider importing this struct
    |
4   | use crate::player::rusty_backend::cpal::OutputStreamTimestamp;
    |
help: if you import `OutputStreamTimestamp`, refer to it directly
    |
775 -             let timestamp = crate::OutputStreamTimestamp { callback, playback };
775 +             let timestamp = OutputStreamTimestamp { callback, playback };
    |

error[E0412]: cannot find type `StreamInstant` in the crate root
  --> /Users/tyler/.asdf/installs/rust/1.64.0/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/mod.rs:73:20
   |
73 | ) -> Result<crate::StreamInstant, BackendSpecificError> {
   |                    ^^^^^^^^^^^^^ not found in the crate root
   |
help: consider importing this struct
   |
3  | use crate::player::rusty_backend::cpal::StreamInstant;
   |
help: if you import `StreamInstant`, refer to it directly
   |
73 - ) -> Result<crate::StreamInstant, BackendSpecificError> {
73 + ) -> Result<StreamInstant, BackendSpecificError> {
A used type name is not in scope.

Erroneous code examples:

```
impl Something {} // error: type name `Something` is not in scope

A used type name is not in scope.

Erroneous code examples:

```
impl Something {} // error: type name `Something` is not in scope

// or:

trait Foo {
    fn bar(N); // error: type name `N` is not in scope
}

// or:

fn foo(x: T) {} // type name `T` is not in scope
```

To fix this error, please verify you didn't misspell the type name, you did
declare it or imported it into the scope. Examples:
   |

error[E0433]: failed to resolve: could not find `StreamInstant` in the crate root
  --> /Users/tyler/.asdf/installs/rust/1.64.0/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/mod.rs:80:15
   |
80 |     Ok(crate::StreamInstant::new(secs as i64, subsec_nanos as u32))
   |               ^^^^^^^^^^^^^ not found in the crate root
   |
help: consider importing this struct
   |
3  | use crate::player::rusty_backend::cpal::StreamInstant;
   |
help: if you import `StreamInstant`, refer to it directly
   |
80 -     Ok(crate::StreamInstant::new(secs as i64, subsec_nanos as u32))
80 +     Ok(StreamInstant::new(secs as i64, subsec_nanos as u32))
   |

error[E0412]: cannot find type `SampleRate` in the crate root
  --> /Users/tyler/.asdf/installs/rust/1.64.0/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/mod.rs:84:51
   |
84 | fn frames_to_duration(frames: usize, rate: crate::SampleRate) -> std::time::Duration {
   |                                                   ^^^^^^^^^^ not found in the crate root
   |
help: consider importing this struct
   |
3  | use crate::player::rusty_backend::cpal::SampleRate;
   |
help: if you import `SampleRate`, refer to it directly
   |
84 - fn frames_to_duration(frames: usize, rate: crate::SampleRate) -> std::time::Duration {
84 + fn frames_to_duration(frames: usize, rate: SampleRate) -> std::time::Duration {
   |

Some errors have detailed explanations: E0412, E0422, E0432, E0433.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `termusic` due to 12 previous errors
error: failed to compile `termusic v0.7.3`, intermediate artifacts can be found at `/var/folders/6k/w8ysm2_x1_j2vym7y6cnb_hm0000z8/T/cargo-installjySvkQ`

```

**Comment by tramhao** [comment]
I compile again v0.7.3 with rust v1.64.0 under arch linux and it works. Seems that the cpal package I included need some update with v1.64.0 under macos.

**Comment by tramhao** [comment]
I figured out the problem. I merged cpal into termusic to remove a warning, but the problem is I didn't compile for macos, so the error occurs. I still don't know how to fix it yet.

**Comment by tramhao** [comment]
basically what you need to do, is to change the crate:: to super:: or super::super, depend on the depth of the file.

**Comment by tramhao** [comment]
Please checkout the latest code in master branch. I change the cpal package in macos to the original version. I don't have chance to compile but I think it should work. Thanks.

**Comment by tramhao** [comment]
I use github actions to verify the code, and seems work fine. Please check if it's working on your computer. Thanks.

**Comment by tramhao** [comment]
released in v0.7.4
